src/multivariate/dirichlet.jl

```diff
@@ -375,3 +375,63 @@ function fit_mle(::Type{<:Dirichlet}, P::AbstractMatrix{Float64},
     elogp = mean_logp(suffstats(Dirichlet, P, w))
     fit_dirichlet!(elogp, α; maxiter=maxiter, tol=tol, debug=debug)
 end
+
+## Differentiation
+function ChainRulesCore.frule((_, Δalpha)::Tuple{Any,Any}, DT::Union{Type{Dirichlet{T}}, Type{Dirichlet}}, alpha::AbstractVector{T}; check_args = true) where {T}
+    d = DT(alpha; check_args=check_args)
+    Δalpha = ChainRulesCore.unthunk(Δalpha)
+    ∂alpha0 = sum(Δalpha)
+    ∂lmnB::typeof(∂alpha0) = sum(Δalpha[i] * (SpecialFunctions.digamma(alpha[i]) - SpecialFunctions.digamma(d.alpha0)) for i in eachindex(alpha))
```

Comment:
through the change of type, it seems now that the inference works so the type annotation is not needed anymore.
For the indices I'm not sure I follow, it doesn't seem correct if alpha and Deltaalpha don't have the same indices?

Reply:
I meant that indexing `alpha` and `Deltaalpha` with the same indices `i` is not necessarily correct since one might use e.g. offset indices whereas the other doesn't. So `zip`, or even better `Broadcast.instantiate(Broadcast.broadcasted(...))` with pairwise summation, seems better.